### DISCUSSION TITLE
Opening a dialog panel with a render fragment?

### DISCUSSION THREAD

**Discussion by SDPBC-JSheridan** [microsoft/fluentui-blazor#2185] [Q&A] [answered]
Hi folks, 

Our team is new to Fluent, and we've been struggling a bit with the way that the ShowPanel methods (https://www.fluentui-blazor.net/Panel) work as opposed to the the standard ShowDialog methods (https://www.fluentui-blazor.net/Dialog) which seem to offer a bit more flexability.  

In particular, we were trying to find a way to display a single component within a panel using a render fragment.  I did some digging through the source to see how the ShowDialog methods work and discovered all that the ShowPanel methods seem to be doing is calling ShowDialog and then adding in the DialogType and Alignment parameters to the DialogParameters object. 

https://github.com/microsoft/fluentui-blazor/blob/b4e31141ae42e8b40c430150e80b9b901112b689/src/Core/Components/Dialog/Services/DialogService-Dialog.cs#L49
https://github.com/microsoft/fluentui-blazor/blob/b4e31141ae42e8b40c430150e80b9b901112b689/src/Core/Components/Dialog/Services/DialogService-Dialog.cs#L76

As it turns out it seems we can just use the more flexible ShowDialog methods (including the RenderFragment overload) instead of the ShowPanel methods by simply feeding in the DialogType and Alignment parameters.  

So, my questions are:
Q1 - Is this actually a kosher way of doing this?
Q2 - If (Q1 == "Yes") - Can the panel documentation (https://www.fluentui-blazor.net/Panel) be updated to reflect this?

Thanks kindly,

John Sheridan
Specialist - IT Solutions
School District of Palm Beach County

**Comment by vnbaaij** [comment, marked as answer]
Q1 - Yes, that is a kosher way of doing that. We created the ShowPanel variants as convenience methods to make using panels a bit easier. As you found, it all calls the same code in the end.

Q2 - Yes, we can update those docs to make this clearer

**Reply by SDPBC-JSheridan** [reply]
Thanks so much Vincent for the reply.   This is super helpful! :)

**Reply by vnbaaij** [reply]
I'm adding this to the Panel page:

>Internally, the `ShowPanel` methods call the `ShowDialog` methods. If is possible to directly call these methods and thereby have access to all of the parameters. The `ShowPanel` variants are just convenience methods that make it easier to work with panels.
